hack/versions/pkg/schema/check.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schema
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/GoogleContainerTools/skaffold/hack/versions/pkg/diff"
+	"github.com/GoogleContainerTools/skaffold/hack/versions/pkg/version"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func RunSchemaCheckOnChangedFiles() error {
+	git, err := newGit()
+	if err != nil {
+		return err
+	}
+	changedFiles, err := git.getChangedFiles()
+	if err != nil {
+		return err
+	}
+	var changedConfigFiles []string
+	for _, file := range changedFiles {
+		if strings.Contains(file, "config.go") {
+			changedConfigFiles = append(changedConfigFiles, file)
+		}
+	}
+
+	root, err := ioutil.TempDir("", "skaffold")
+	if err != nil {
+		return err
+	}
+	var filesInError []string
+	baseRef := "master"
+	for _, configFile := range changedConfigFiles {
+		content, err := ioutil.ReadFile(configFile)
+		if err != nil {
+			return errors.Wrapf(err, "reading %s", configFile)
+		}
+		changedFile := path.Join(root, "changed.go")
+		if err = ioutil.WriteFile(changedFile, content, 0666); err != nil {
+			return errors.Wrapf(err, "writing changed version of %s", configFile)
+		}
+
+		content, err = git.getFileFromRef(configFile, baseRef)
+		if err != nil {
+			return err
+		}
+		baseFile := path.Join(root, "base.go")
+		if err = ioutil.WriteFile(baseFile, content, 0666); err != nil {
+			return errors.Wrapf(err, "writing master version of %s", configFile)
+		}
+
+		diff, err := diff.CompareGoStructs(baseFile, changedFile)
+		if err != nil {
+			return errors.Wrapf(err, "failed to compare go files %s vs %s", baseFile, changedFile)
+		}
+
+		isLatest := strings.Contains(configFile, "latest")
+		if diff == "" {
+			continue
+		}
+		if !isLatest {
+			filesInError = append(filesInError, configFile)
+			continue
+		}
+
+		logrus.Infof("structural changes in latest config, checking on Github if latest is released...")
```

Comment:
This is incorrect, we actually are checking for file changes in all version config.go and not just latest.

Reply:
see L83